scipy/linalg/_basic.py

```diff
@@ -1435,14 +1436,15 @@ def pinv(a, atol=None, rtol=None, return_rank=False, check_finite=True,
     t = u.dtype.char.lower()
     maxS = np.max(s)
 
-    if rcond or cond:
+    if rcond is not _NoValue or cond is not _NoValue:
         warn('Use of the "cond" and "rcond" keywords are deprecated and '
-             'will be removed in future versions of SciPy. Use "atol" and '
+             'will be removed in SciPy 1.14.0. Use "atol" and '
              '"rtol" keywords instead', DeprecationWarning, stacklevel=2)
 
     # backwards compatible only atol and rtol are both missing
-    if (rcond or cond) and (atol is None) and (rtol is None):
-        atol = rcond or cond
+    if ((rcond is not _NoValue or cond is not _NoValue)
+        and (atol is None) and (rtol is None)):
+        atol = rcond if rcond is not _NoValue else cond
```

Comment:
I don't think this is correct for all combinations of `None`,  `_NoValue`, <else>.

Reply:
Thanks good catch